docs/source/examples/sentiment.rst

```diff
@@ -0,0 +1,352 @@
+First, we explain the training method of Recursive Neural Networks without
+mini-batch processing. After that, as an advanced story, we explain the training
+method of a mini-batched Recursive Neural Network.
+
+In this tutorial, we will learn the following:
+
+#. What is a Recursive Neural Network?
+#. Implementation of sentiment analysis by Recursive Neural Networks using Chainer
+
+    * Training method of a Recursive Neural Network without mini-batches
+    * Training method of a mini-batched Recursive Neural Network
+
+1. What is a Recursive Neural Network? 
+=======================================
+
+A Recursive Neural Network is a Recurrent Neural Networks that extends to
+a tree structure. As both networks are often written as RNN, we need to be
+careful which one we are referring to. RNN usually refers to the Recurrent
+Neural Networks, but in natural language processing it sometimes
+refers to Recursive Neural Networks.
+
+Recursive Neural Networks use a tree structure with a fixed number of branches.
+In the case of a binary tree, the hidden state vector of the current node is
+computed from the hidden state vectors of the left and right child nodes,
+as follows:
+
+.. math::
+    {\bf h}_P = a \left( {\bf W} 
+    \left[ \begin{array}{l}
+        {\bf h}_L \\
+        {\bf h}_R \\
+    \end{array} \right]
+    + {\bf b} \right)
+
+.. image:: ../../image/sentiment/rnn.png
+
+This operation is sequentially calculated from the leaf nodes toward the root node.
+Recursive Neural Networks are expected to express relationships between long-distance
+elements compared to Recurrent Neural Networks, because the depth is enough with
+:math:`log_2(T)` if the element count is :math:`T`.
+
+2. Implementation of sentiment analysis by Recursive Neural Networks
+=====================================================================
+
+2.1 Preparation of training data
+---------------------------------
+
+In this tutorial, we will use the training data which are preprocessed by 
+`chainer/examples/sentiment/download.py <https://github.com/chainer/chainer/blob/master/examples/sentiment/download.py>`_.
+
+.. literalinclude:: ../../../examples/sentiment/download.py
+   :language: python
+   :caption: download.py
+
+Let's run the following codes, download the necessary training data and unzip it.
+
+.. code-block:: console
+
+    $ ./download.py
+
+Let's execute the following command and check if the training data have been prepared.
+
+.. code-block:: console
+
+    $ ls trees
+    dev.txt  test.txt  train.txt
+
+Let's look at the first line of test.txt and see how each sample is written.
+
+.. code-block:: console
+
+    $ head trees/dev.txt -n1
+    (3 (2 It) (4 (4 (2 's) (4 (3 (2 a) (4 (3 lovely) (2 film))) (3 (2 with) (4 (3 (3 lovely) (2 performances)) (2 (2 by) (2 (2 (2 Buy) (2 and)) (2 Accorsi))))))) (2 .)))
+
+As displayed above, each sample is defined by a tree structure.
```

Comment:
Please refer about a parser. In NLP field, such kind of trees are called "parse tree".
This structure is generated by Stanford parser according to the paper. https://nlp.stanford.edu/software/lex-parser.shtml